packages/core/EventManager.ts

```diff
@@ -338,33 +341,45 @@ export default class EventManager {
   private async createAllCalendarEvents(event: CalendarEvent) {
     let createdEvents: EventResult<NewCalendarEventType>[] = [];
     if (event.destinationCalendar && event.destinationCalendar.length > 0) {
-      for (const destination of event.destinationCalendar) {
+      // Since GCal pushes events to multiple calendars we only want to create one event per booking
+      let gCalAdded = false;
+      const destinationCalendars: DestinationCalendar[] = event.destinationCalendar.reduce(
+        (destinationCals, cal) => {
+          if (cal.integration === "google_calendar") {
+            if (gCalAdded) {
+              return destinationCals;
+            } else {
+              gCalAdded = true;
+              destinationCals.push(cal);
+            }
+          } else {
```

Comment:
This should be handled in google calendar App. Also, are we sure that no other calendar app has the same behaviour as Google Calendar as that might still be having duplicate events.
cc @joeauyeung @alannnc

Reply:
I was thinking about that but the GCal app is called on an individual calendar level so it wouldn't know if other calendars have already created the event.

I tested this with Outlook since it's our second biggest calendar and it wasn't having the same issue.

Reply:
We have got to change the abstraction for CalendarService then. We could start by having createEvent accept another argument(destinationCalendars) that can for now be used only by Google Calendar app.